### PR TITLE
Issue-1167: Replace application.properties by yaml equivalent.

### LINKED ISSUE #1167: Replace application.properties by yaml equivalent
# Task Description

See also https://docs.spring.io/spring-boot/docs/current/reference/html/boot-features-external-config.html#boot-features-external-config-yaml

# Acceptance Test

* [ ] Building the code and running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [ ] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [ ] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Task Relationships

This task is a follow-up of #965 (#1056).

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

### PR DESCRIPTION
**Pull request by ptirador**
Fixes #1167.

- Replaced `application.properties` with `application.yaml`.

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

### CODE REVIEW

File: strongbox-resources/strongbox-storage-api-resources/pom.xml [hunk 1]
```diff
@@ -45,6 +45,16 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-resources-plugin</artifactId>
+                <configuration>
+                    <useDefaultDelimiters>false</useDefaultDelimiters>
+                    <delimiters>
+                        <delimiter>@{*}</delimiter>
```
Comment (carlspring) [inline strongbox-resources/strongbox-storage-api-resources/pom.xml]: Just out of curiousity and for my better understanding, why are you changing the delimiter? Is the `$` sign a special character in YAML?

Comment (ptirador) [reply strongbox-resources/strongbox-storage-api-resources/pom.xml]: This delimiter is used to make reference to a maven POM property and avoid conflicts with the "${*}" deliimter. [Link](https://docs.spring.io/spring-boot/docs/current/reference/html/howto-properties-and-configuration.html).

**Comment by carlspring** [comment]
@fuss86 : Thanks for your help!

@ptirador : Is this now ready for merging?

**Comment by ptirador** [comment]
@fuss86: Thank so mucn for you help.
@carlspring: Yes, it's ready.

**Pull request merged**